crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1749,6 +1977,62 @@ impl<T: R2D2Connection> IndexerStore for PgIndexerStore<T> {
             .await
     }
 
+    async fn prune_epoch(&self, epoch: u64) -> Result<(), IndexerError> {
+        let (mut min_cp, max_cp) = match self.get_checkpoint_range_for_epoch(epoch)? {
+            (min_cp, Some(max_cp)) => Ok((min_cp, max_cp)),
+            _ => Err(IndexerError::PostgresReadError(format!(
+                "Failed to get checkpoint range for epoch {}",
+                epoch
+            ))),
+        }?;
+
+        // NOTE: for disaster recovery, otherwise get_transaction_range_for_checkpoint would fail,
+        // as cp_tx table has pruned the cp already.
+        let min_prunable_cp = self.get_min_prunable_checkpoint()?;
```

Comment:
maybe a comment explaining why this particular order (to minimize any inconsistencies when pruning, culminating in pruning the epoch table)